src/Event/Http/Psr7Bridge.php

```diff
@@ -23,22 +23,33 @@ final class Psr7Bridge
      */
     public static function convertRequest(HttpRequestEvent $event, Context $context): ServerRequestInterface
     {
+        $headers = $event->getHeaders();
+
         [$files, $parsedBody] = self::parseBodyAndUploadedFiles($event);
+        [$user, $password] = self::parseBasicAuthorization($headers);
 
-        $server = [
-            'SERVER_PROTOCOL' => $event->getProtocolVersion(),
+        $server = array_filter([
+            'CONTENT_LENGTH' => $headers['content-length'][0] ?? null,
+            'CONTENT_TYPE' => $event->getContentType(),
+            'DOCUMENT_ROOT' => getcwd(),
+            'QUERY_STRING' => $event->getQueryString(),
             'REQUEST_METHOD' => $event->getMethod(),
+            'SERVER_NAME' => $event->getServerName(),
+            'SERVER_PORT' => $event->getServerPort(),
+            'SERVER_PROTOCOL' => $event->getProtocol(),
+            'PATH_INFO' => $event->getPath(),
+            'HTTP_HOST' => $headers['host'] ?? null,
+            'REMOTE_ADDR' => $event->getSourceIp(),
+            'REMOTE_PORT' => $event->getRemotePort(),
             'REQUEST_TIME' => time(),
             'REQUEST_TIME_FLOAT' => microtime(true),
-            'QUERY_STRING' => $event->getQueryString(),
-            'DOCUMENT_ROOT' => getcwd(),
             'REQUEST_URI' => $event->getUri(),
-            'REMOTE_ADDR' => $event->getSourceIp(),
-        ];
+            'PHP_AUTH_USER' => $user,
+            'PHP_AUTH_PW' => $password,
+        ], fn ($value) => ! is_null($value));
```

Comment:
I _think_ `array_filter` works without a callback too, and it might do what you want (remove empty values). That would solve the php-cs error as well.

Reply:
done